crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -381,6 +511,120 @@ impl Provider for KvRedisProvider {
         Ok(())
     }
 
+    async fn receive_link_config_as_source(
+        &self,
+        LinkConfig {
+            target_id,
+            config,
+            secrets,
+            link_name,
+            ..
+        }: LinkConfig<'_>,
+    ) -> anyhow::Result<()> {
+        let url = secrets
+            .keys()
+            .find(|k| k.eq_ignore_ascii_case(CONFIG_REDIS_URL_KEY))
+            .and_then(|url_key| config.get(url_key))
+            .or_else(|| {
+                warn!("Redis connection URLs can be sensitive. Consider using secrets to pass this value.");
+                config.keys()
+                    .find(|k| k.eq_ignore_ascii_case(CONFIG_REDIS_URL_KEY))
+                    .and_then(|url_key| config.get(url_key))
+            })
+            .map_or(DEFAULT_CONNECT_URL, |v| v);
+
+        let client = match redis::Client::open(url.to_string()) {
+            Ok(client) => {
+                info!(url, "Established link");
+                client
+            }
+            Err(err) => {
+                warn!(?err, "Failed to create Redis client for target [{target_id}]. Key-value operations will fail.");
+                bail!("Failed to create Redis client");
+            }
+        };
+
+        let mut conn = client.get_connection_manager().await.map_err(|e| {
+            error!("Failed to get async connection: {e}");
+            anyhow::anyhow!("Failed to get async connection: {}", e)
+        })?;
+
+        // Set keyspace notifications
+        redis::cmd("CONFIG")
+            .arg("SET")
+            .arg("notify-keyspace-events")
+            .arg("KE")
+            .query_async::<_, ()>(&mut conn)
+            .await
+            .map_err(|e| {
+                error!("Failed to set keyspace notifications: {e}");
+                anyhow::anyhow!("Failed to set keyspace notifications: {}", e)
+            })?;
+
+        let mut sources = self.sources.write().await;
+        sources.insert((target_id.to_string(), link_name.to_string()), conn);
+
+        // Start watching for keyspace events
+        let client_clone = client.clone();
+        let self_clone = self.clone();
+        tokio::spawn(async move {
```

Comment:
It's probably a good idea to store this task as a part of the sources so that we can kill the task later on if the component isn't linked anymore

Also, I think we only want to spawn this task if the [`LinkConfig { wit_metadata }`](https://docs.rs/wasmcloud-provider-sdk/0.11.0/wasmcloud_provider_sdk/struct.LinkConfig.html#structfield.wit_metadata) interface contains `watcher`, otherwise we don't need to watch the keys